packages/components/modal/src/use-modal.ts

```diff
@@ -129,6 +129,12 @@ export function useModal(originalProps: UseModalProps) {
     },
   });
 
+  useEffect(() => {
+    if (isOpen) {
+      onOpenChange?.(isOpen);
+    }
+  }, [isOpen]);
```

Comment:
It looks like the issue might be with `useOverlayTriggerState` not properly triggering `onOpenChange`. Could you leave a comment about that? It would also be helpful if you could file an issue upstream if possible.

https://github.com/nextui-org/nextui/blob/8a33eabb2583202fcc8fbc33e8f2ed23bb45f1a4/packages/components/modal/src/use-modal.ts#L121-L130

Reply:
Is there a possibility that this change might cause `onOpenChange` to be called multiple times?

Reply:
> It looks like the issue might be with `useOverlayTriggerState` not properly triggering `onOpenChange`. Could you leave a comment about that? It would also be helpful if you could file an issue upstream if possible.
> 
> https://github.com/nextui-org/nextui/blob/8a33eabb2583202fcc8fbc33e8f2ed23bb45f1a4/packages/components/modal/src/use-modal.ts#L121-L130

Basically, `useOverlayTriggerState` fires `onOpenChange` when its internal `onClose` and `onOpen` are called. 
But in our case `onOpen` is coming from `useDisclosure` and `onClose`  is the internal function of `useOverlayTriggerState`  which is passed while mounting modal.
It has its own state `isOpen`, so its doesn't react automatically to outside state, as per my understanding.

> https://github.com/nextui-org/nextui/blob/8a33eabb2583202fcc8fbc33e8f2ed23bb45f1a4/packages/components/modal/src/use-modal.ts#L205

Reply:
> Is there a possibility that this change might cause `onOpenChange` to be called multiple times?

Don't think so , because I am only calling when the `onOpen` changes to `true`.  However, one issue could be if we call the `onClose` fxn which is return by `useDisclosure`, than same problem will arise, `onOpenChange` will not be called.

Reply:
Also, I thought if we can somehow expose the both the fxn of `useOverlayTriggerState`  in useDisclosure, than that will be very neat fix. But that is not possible unless the modal is mounted.

Reply:
Oh, I see. Thank you for the clarification.